oe_contact_forms.module

```diff
@@ -6,3 +6,350 @@
+    '#description' => t("Check this box if you'd like to make this a corporate form."),
+    '#default_value' => $is_corporate_form,
+    '#ajax' => [
+      'callback' => '_oe_contact_forms_is_corporate_form_ajax_callback',
+      'wrapper' => 'corporate-fields-wrapper',
+    ],
+  ];
+  $form['corporate_fields'] = [
+    '#type' => 'container',
+    '#attributes' => [
+      'id' => ['corporate-fields-wrapper'],
+    ],
+  ];
+
+  if ($is_corporate_form) {
+    $form['corporate_fields']['topics_fieldset'] = [
+      '#type' => 'fieldset',
+      '#title' => t('Topics'),
+      '#prefix' => '<div id="topics-fieldset-wrapper">',
+      '#suffix' => '</div>',
+      '#description' => t("Please select the topics and the associated email addresses the users can choose when submitting the form. For multiple email addresses, please separate them with a comma."),
+    ];
+
+    $topics = $contact_form->getThirdPartySetting('oe_contact_forms', 'topics', []);
+    $num_topics = $form_state->get('num_topics');
+    $n = count($topics);
+    $topics_limit_validation_errors = [];
+
+    // No values added yet.
+    if ($num_topics === NULL && $n === 0) {
+      $num_topics = 1;
+    }
+
+    // Any saved values.
+    if ($num_topics === NULL && $n > 0) {
+      $num_topics = $n;
+    }
+
+    $form_state->set('num_topics', $num_topics);
+
+    for ($i = 0; $i < $num_topics; $i++) {
+      if (isset($topics[$i])) {
+        $topic_name = $topics[$i]['topic_name'];
+        $topic_email_address = $topics[$i]['topic_email_address'];
+        $topics_limit_validation_errors[] = [
+          'corporate_fields',
+          'topics_fieldset',
+          'group',
+          $i,
+          'topic_name',
+        ];
+        $topics_limit_validation_errors[] = [
+          'corporate_fields',
+          'topics_fieldset',
+          'group',
+          $i,
+          'topic_email_address',
+        ];
+      }
+      $form['corporate_fields']['topics_fieldset']['group'][$i]['topic_name'] = [
+        '#type' => 'textfield',
+        '#title' => t('Topic name'),
+        '#required' => TRUE,
+        '#default_value' => $topic_name ?? '',
+      ];
+      $form['corporate_fields']['topics_fieldset']['group'][$i]['topic_email_address'] = [
+        '#type' => 'textfield',
+        '#title' => t('Topic email address(es)'),
+        '#required' => TRUE,
+        '#default_value' => $topic_email_address ?? '',
+      ];
+    }
+
+    $form['corporate_fields']['topics_fieldset']['actions'] = [
+      '#type' => 'actions',
+    ];
+    $form['corporate_fields']['topics_fieldset']['actions']['add_topic'] = [
+      '#type' => 'submit',
+      '#value' => t('Add topic'),
+      '#submit' => ['_oe_contact_forms_add_one_callback'],
+      '#ajax' => [
+        'callback' => '_oe_contact_forms_add_more_callback',
+        'wrapper' => 'topics-fieldset-wrapper',
+      ],
+      '#limit_validation_errors' => $topics_limit_validation_errors,
```

Comment:
The limit validation error doesn't work. i am allowed to add more topic sets without adding a label or email.

Reply:
fixed